packages/eslint-plugin/tests/rules/no-misused-promises.test.ts

```diff
@@ -870,5 +898,32 @@ it('', async () => {});
         },
       ],
     },
+    {
+      code: `
+console.log({ ...Promise.resolve({ key: 42 }) });
```

Comment:
A few more failing cases to consider:

* `...(condition && Promise.resolve({ key: 42 }))`
* `...(condition ? Promise.resolve({ key: 42 }) : {})`
* Array spreads in general

Reply:
Good point, I will add support for logical operators. I think array spreads are not relevant, though — `Promise` doesn't implement the iterator protocol, so it can't be spread in an array. Typescript already warns about that.

Reply:
True, good point! I think it'd be useful to have a valid test case just to make sure `[...await ...]` is happy still.

Reply:
I'm not familiar with Typescript AST parsing, so maybe you could help me here, conceptually. The type of logical expressions is not thenable, but we can get the type that the expression will resolve to. However, if we get e.g. `false | Promise<Foo>`, then that's still not thenable (?). It appears to me that in the case of conditionals, we check that all parts of the union are thenable. For spreading, would it make sense to check that _some_ parts are thenable (since it's still likely a mistake)?

Reply:
Hmm, good questions... I agree, if there's a type union, then checking that _some_ parts are thenable seems like the right approach. +1 that it's still likely a mistake.